src/core/assertions.js

```diff
@@ -92,36 +97,57 @@ exports.assertObjectMatchSpec = (object, spec, exact = false) => {
         const currentValue = _.get(object, field)
         const expectedValue = Cast.value(value)
 
-        switch (matcher) {
-            case 'match':
-            case 'matches':
-                expect(
-                    currentValue,
-                    `Property '${field}' (${currentValue}) does not match '${expectedValue}'`
-                ).to.match(new RegExp(expectedValue))
-                break
+        if (matcher.indexOf('matches') !== -1 || matcher.indexOf('match') !== -1) {
```

Comment:
why not checking for `0`? (the same goes for others)

Reply:
and if it matches `match` it matches `matches` :) no need for the `OR`

Reply:
I see your point for `0`. You're right for the second as well. It's just that I'm more accustomed to `-1`. `0` in this case is stricter. 

But I find `indexOf`  somehow to permissive in the end as it will also match `matchess`, though the end user made a typo. Shouldn't we use a regex to be stricter?

Reply:
yes, you're right, it's also better for DX as user get immediate feedback on potential typos, it's worth the extra cost

Reply:
Works for me. I create regex for each matcher